core/src/test/java/org/opensearch/sql/expression/operator/convert/TypeCastOperatorTest.java

```diff
@@ -361,4 +372,27 @@ void castToDatetime() {
     assertEquals(new ExprDatetimeValue("2012-08-07 00:00:00"), expression.valueOf(null));
   }
 
+  @Test
+  void typeof() {
+    assertEquals("UNDEFINED", typeofGetValue(ExprNullValue.of()));
```

Comment:
is it possible to test UNKNOWN?

Reply:
Yes, see `typeof(NULL)` in integration tests